src/PdfSharper/Pdf.IO/Lexer.cs

```diff
@@ -211,10 +247,32 @@ public byte[] ReadStream(int length)
         /// </summary>
         public String ReadRawString(int position, int length)
         {
-            _pdfSteam.Position = position;
+            Position = position;
             byte[] bytes = new byte[length];
-            _pdfSteam.Read(bytes, 0, length);
-            return PdfEncoders.RawEncoding.GetString(bytes, 0, bytes.Length);
+
+            int pos = _idxChar - _bufferOffset;
+            int read = 0;
+            do
+            {
+                int available = Math.Min(length - read, _bufferLength - pos);
+                Buffer.BlockCopy(_byteBuffer, pos, bytes, read, available);
+                read += available;
+
+                if (read < length)
+                {
+                    _pdfStream.Position = _bufferOffset + _bufferLength;
```

Comment:
Duplicate of above and roughly a dupe of FillBuffer(). Should be possible to use the same method for each.